internal/controlplane/handlers_oauth.go

```diff
@@ -56,6 +58,11 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 		return nil, providerError(err)
 	}
 
+	if !slices.Contains(provider.AuthFlows, db.AuthorizationFlowOauth2AuthorizationCodeFlow) {
```

Comment:
Can we use `provider.SupportsAuthFlow(...)` here?  I think this is the same comment jakub had down below.

Ideally, `provider` returned from `getProviderFromRequestOrDefault` would be either an actual Provider instance, or at least a ProviderBuilder instance.  But that's a larger refactor.